metricshub-jawk-extension/src/main/java/org/sentrysoftware/metricshub/extension/jawk/MetricsHubExtensionForJawk.java

```diff
@@ -0,0 +1,334 @@
+	 * @param args The array of arguments to use to create the {@link WmiSource}.
+	 * @return The table result from the execution of the {@link WmiSource}.
+	 */
+	private String executeWmiRequest(final Object[] args) {
+		if ((args[0] instanceof AssocArray argsAssocArray)) {
+			return executeSource(
+				WmiSource
+					.builder()
+					.type("wmi")
+					.query(toAwkString(argsAssocArray.get("query")))
+					.namespace(toAwkString(argsAssocArray.get("namespace")))
+					.forceSerialization(toAwkString(argsAssocArray.get("forceSerialization")).equals("true"))
+					.build()
+			);
+		} else {
+			log.warn("Hostname {} - Jawk: executeWmiRequest(): expected Association array.", hostname);
+			return "";
+		}
+	}
+
+	/**
+	 * Execute a {@link Source} through the context.
+	 * @param source The {@link Source} to execute.
+	 * @return The table result from the execution of the {@link Source}.
+	 */
+	private String executeSource(final Source source) {
+		final SourceTable sourceTableResult = source.accept(sourceProcessor);
+
+		return sourceTableResult != null && !sourceTableResult.isEmpty() ? sourceTableResult.getRawData() : "";
+	}
+
+	/**
+	 * Execute a {@link Json2Csv} compute on the current source through the context.
+	 * @param args The array of arguments to use to create the {@link Json2Csv} compute.
+	 * @return The table result from the execution of the compute.
+	 */
+	private String executeJson2csv(final Object[] args) {
+		if ((args[0] instanceof AssocArray argsAssocArray)) {
+			try {
+				return ClientsExecutor
+					.executeJson2Csv(
+						toAwkString(argsAssocArray.get("jsonSource")),
+						toAwkString(argsAssocArray.get("entryKey")),
+						toAwkListString(argsAssocArray.get("properties")),
+						toAwkString(argsAssocArray.get("separator")),
+						hostname
+					)
+					.strip();
+			} catch (Exception exception) {
+				log.error(
+					"Hostname {} - Json2Csv Operation has failed. Errors:\n{}\n",
+					hostname,
+					StringHelper.getStackMessages(exception)
+				);
+				return "";
+			}
+		} else {
+			log.warn("Hostname {} - Jawk: executeJson2csv(): expected Association array.", hostname);
+			return "";
+		}
+	}
+
+	/**
+	 * Convert a Jawk variable to a {@link List} of {@link String}, assuming the Jawk variable contains a list of parameters separated by ';'.
+	 *
+	 * @param arg The Jawk variable to convert.
+	 * @return The Jawk variable converted to a {@link List} of {@link String}.
+	 */
+	private List<String> toAwkListString(final Object arg) {
+		final String stringArg = toAwkString(arg);
+		return Arrays.asList(stringArg.split(";"));
+	}
+
+	/**
+	 * Return the value of the variable in parameter if it exists.
+	 * @param arg The name of the variable to retrieve.
+	 * @return The value of the variable.
+	 */
+	private String getVariable(final Object[] args) {
+		final String var = toAwkString(args[0]);
+		final String res = sourceProcessor.getTelemetryManager().getHostConfiguration().getConnectorVariables().get(var);

```

Comment:
The `ConnectorVariables` in `HostConfiguration` is poorly designed. The `HostConfiguration.connectorVariables` should be:

```java
private Map<String, ConnectorVariables> variables = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
```

Here, the `String` key in the variables map represents the connector identifier. This change consists in two key updates:

1. Modify `ConfigHelper.buildHostConfiguration` in the `metricshub-agent` to set `resourceConfig.variables` within the `HostConfiguration`.
2. `MetricsHubExtensionForJawk` must be instantiated for each host and connector ID, as the `connectorId` field is needed inside it. (Like you did the first time)

@bertysentry fyi.